soroban-env-host/src/vm/module_cache.rs

```diff
@@ -73,43 +180,96 @@ impl ModuleCache {
                                 v1.cost_inputs.metered_clone(host.as_budget())?,
                             ),
                         };
-                        self.parse_and_cache_module(host, hash, code, code_cost_inputs)?;
+                        self.parse_and_cache_module(
+                            host,
+                            host.get_ledger_protocol_version()?,
+                            hash,
+                            code,
+                            code_cost_inputs,
+                        )?;
                     }
                 }
             }
         }
         Ok(())
     }
 
-    pub fn parse_and_cache_module(
+    pub fn parse_and_cache_module_simple<Ctx: CompilationContext>(
         &mut self,
-        host: &Host,
+        context: &Ctx,
+        curr_ledger_protocol: u32,
+        wasm: &[u8],
+    ) -> Result<(), HostError> {
+        let contract_id = Hash(crate::crypto::sha256_hash_from_bytes_raw(
+            wasm,
+            context.as_budget(),
+        )?);
+        self.parse_and_cache_module(
+            context,
+            curr_ledger_protocol,
+            &contract_id,
+            wasm,
+            VersionedContractCodeCostInputs::V0 {
+                wasm_bytes: wasm.len(),
+            },
+        )
+    }
+
+    pub fn parse_and_cache_module<Ctx: CompilationContext>(
```

Comment:
Nit: this can be `pub (crate) fn`, so is `add_stored_contracts`

Reply:
No, a major point of this change is that the module cache can now be (publicly) operated by stellar-core before and after a host exists.